include/deal.II/base/work_stream.h

```diff
@@ -1008,7 +1008,7 @@ namespace WorkStream
       {
         // need to copy the sample since it is marked const
         ScratchData scratch_data = sample_scratch_data;
-        CopyData    copy_data    = sample_copy_data; // NOLINT
+        auto        copy_data    = CopyData(sample_copy_data);
```

Comment:
That's unfortunate -- because the variable is most definitely modified in this piece of code: we are calling the `worker` function just a couple of lines below. Is it warning about a particular instantiation of this function, or about the general template?

Reply:
When compiling the library this appears for
- `source/numerics/matrix_creator.cc`
- `source/numerics/matrix_creator_inst2.cc`
- `source/numerics/matrix_creator_inst3.cc`

Reply:
Oh well. I tend to think that reading the new form obfuscates the code and invites someone to just change this back to the old form given that it's so much more obvious. On the other hand, there are probably only three people anyway who understand that code, so maybe we just shouldn't care.